lxd/response/response.go

```diff
@@ -28,18 +31,29 @@ func Init(d bool) {
 
 // Response represents an API response.
 type Response interface {
-	Render(w http.ResponseWriter) error
+	Render(w http.ResponseWriter, r *http.Request) error
 	String() string
 }
 
+// UseCallbackFunction retrieves a callback function from the request context and calls it.
+// The callback function is used to mark the request as completed for the API metrics.
+func UseCallbackFunction(request *http.Request, result metrics.RequestResult) {
```

Comment:
Having this in the `response` package seems odd when the related context key is defined in the `request` package. Suggest moving it there (and separating this into its own commit too).

Also there seems to be some tension between tightly coupling this to metrics and keeping it as a generic callback mechanism, because the naming of the context key and the function is generic, but the callback function signature takes a `metrics.RequestResult` tightly coupling it to metrics (as well as the internal logging mentioning metrics too).

If it will be tightly coupled to metrics, which is fine, then lets make it clear what its for in the rest of the naming.

Suggest:
```
request.MetricsCallback()
```

And the context key being called `lxdRequest.MetricsCallbackFunc`